### PR TITLE
Remove unsupported backends

### PR DESCRIPTION
**Pull request by ldmberman**
- Removed the RethinkDB backend;
- removed the change feed functionality;
- removed the `backend/admin.py` interface;
- moved the MongoDB backend implementation completely to `localmongodb`;
- removed the `mongodb-ssl` backend;
- cleaned the `query.py` interface up;
- re-enabled `backend/test_connection.py`, `backend/test_generics.py`, `backend/test_utils.py`;
- fixed the pytest bdb and abci marks descriptions;
- removed genesis-related fixtures;
- made `localmongodb` the default backend; updated the docs to not mention the `--database-backend` option for the pytest command;
- removed multipipes from setup.py.

This is part of the BigchainDB 2.* migration cleanup.

### CODE REVIEW

**Comment by ttmc** [comment]
In BigchainDB 2.0, there's still a genesis block (i.e. the block with height 0), so one can still do things with a genesis block in BigchainDB 2.0.

@shahbazn - You did some stuff with the genesis block code lately. Could you take a look at the "genesis" related changes in this PR?

**Comment by ttmc** [comment]
Travis failed when it tried to build some auto-generated docs about `bigchaindb.backend.changefeed` (which is gone now):

```text
Warning, treated as error:
autodoc: failed to import module 'bigchaindb.backend.changefeed'; the following exception was raised:
No module named 'bigchaindb.backend.changefeed'
```

**Comment by ldmberman** [comment]
@ttmc I've updated the backends doc appendix, could you, please, check?

**Comment by ttmc** [comment]
@ldmberman I checked out this branch on my local machine and Sphinx built all the docs (three sets) with no problems. Also, the Travis tests (including docs building tests) are passing. Good!

**Comment by ldmberman** [comment]
@ttmc yes, I checked that too. Can you, please, look if the docs themselves look good to you?

File: bigchaindb/backend/localmongodb/connection.py [hunk 1]
```diff
@@ -1,5 +1,268 @@
...
+        except pymongo.errors.ConfigurationError as exc:
+            raise ConfigurationError from exc
+
+
+MONGO_OPTS = {
+    'socketTimeoutMS': 20000,
+}
+
+
+def initialize_replica_set(host, port, connection_timeout, dbname, ssl, login,
```
Comment (kansi) [inline bigchaindb/backend/localmongodb/connection.py]: @vrde IIRC there we disscussions around removing replica set? If yes, then should we do that in this PR?

**Comment by ttmc** [comment]
@ldmberman Yes, that page (Server docs - Appendices - Database Backend Interfaces) _looks_ okay, but there are still many issues, e.g.

- There are a couple of references to the "backlog" but there is no backlog with BigchainDB 2.0. The backlog used to be a BigchainDB-managed table/collection where transactions would wait until they got included in a block, but now the equivalent thing (the "mempool") is managed by Tendermint.
- We use the word "table" very often, but since MongoDB is the only backend option now, we could just use the word "collection". This is a matter of taste and could be ignored.

Maybe cleanup of "backlog" and "table" references could be done in a new PR. This PR is getting too big. Big is bad, because review takes long and slows us down.

**Comment by ldmberman** [comment]
@ttmc I have not noticed these backlog references, thanks! For example, we can create a dedicated PR to remove all traces of backlog in the codebase.

File: setup.py [hunk 2]
```diff
@@ -83,7 +83,6 @@ def check_setuptools_features():
     'flask-restful~=0.3.0',
     'requests~=2.9',
     'gunicorn~=19.0',
-    'multipipes~=0.1.0',
```
Comment (ttmc) [inline setup.py]: I think you can also remove the `'rethinkdb~=2.3',` line in `setup.py`.

**Comment by ttmc** [comment]
Some tests are failing on Travis, such as:

tests/web/test_transactions.py::test_post_create_transaction_with_language[danish-202-False] FAILED

**Comment by ttmc** [comment]
Farewell, RethinkDB!

**Pull request merged**